scrc/preprocessors/extractors/spider_specific/judgment_extracting_functions.py

```diff
@@ -157,6 +157,72 @@ def BS_Omni(rulings: str, namespace: dict) -> Optional[List[Judgment]]:
+    all_judgment_markers = {
+        Language.DE: {
+            Judgment.APPROVAL: [r'Gutheissung der Beschwerde', r'Bejahung der Beschwerdelegimitation'],
+            Judgment.PARTIAL_APPROVAL: [r'Teilweise Gutheissung der Beschwerde'],
+            Judgment.DISMISSAL: [r'Abweisung der Beschwerde', 
+                                r'Der Anzeige wird keine Folge gegeben', 
+                                r'Verneinung der Beschwerdelegimitation', 
+                                r'Abweisung der Verwaltungsgerichtsbeschwerde', 
+                                r'Abweisung der [Vv]erwaltungsrechtlichen Klage',
+                                r'Abweisung des Gesuches um Wiederherstellung der Frist',
+                                r'In concreto Abweisung der Berufung der Dienstbarkeitsbelasteten'],
+            Judgment.WRITE_OFF: [r'Abschreibung der Beschwerde vom Geschäftsprotokoll ']
+        }
+    }
+
+    # In canton UR, de only
+    if namespace['language'] != Language.DE:
+        raise ValueError(f'This function is only implemented for {Language.DE} so far.')
+
+    def getJudgments(text: str, judgment_markers, debug=False) -> Optional[List[Judgment]]:
+        """
+        Get the judgment outcomes based on a regex dictionary for a given section string.
+        :param text:               the text string of the section, usually Section.RULINGS
+        :param judgment_markers:   the regex dicttionary for the different judgment outcomes
+        :param debug:              if set to True, prints debug output to console
+        :return:                   the list of judgment outcomes
+        """
+        
+        judgments = []
+        for lang in judgment_markers:
+            for judg in judgment_markers[lang]:
+                for reg in (judgment_markers[lang])[judg]:
+                    matches = re.finditer(reg, text, re.MULTILINE)
+                    for num, match in enumerate(matches, start=1):
+                        from_, to_, match_text  = match.start(), match.end(), match.group()
+                        judgments.append(judg)
+                        if debug:
+                            print(f'{judg} ("{match.group()}") at {to_/len(text):.1%} of the section.')
+                        
+        return judgments
+    
+    # find all judgments in the rulings
+    judgments = getJudgments(rulings, all_section_markers)
+    
+    # validate
+    if len(judgments) > 1:
+        # TODO: using from_ and to_ position of judgment, check the following cases
+        #   1) a match is included in another --> discard the one included
+        #   2) two matches overlap --> problem, check 
+        #   3) no judgment found --> update the all_judgment_markers 
+        raise ValueError(f"Found several judgments for the rulings {namespace['date']} . Please check!")
+    if len(judgments) == 0:
+        raise ValueError(f"Found no judgment for the rulings {namespace['date']}. Please check!")
```

Comment:
Why do you just log the date here?

Reply:
Yep, url makes much more sense. Changed below.